substation-diagram/substation-diagram-core/src/main/java/com/powsybl/substationdiagram/svg/SVGWriter.java

```diff
@@ -494,23 +517,59 @@ private void drawLabel(String str, boolean rotated, double xShift, double yShift
 
     private boolean canInsertComponentSVG(Node node) {
         return layoutParameters.isShowInternalNodes() ||
-                (!node.isFictitious() && node.getType() != Node.NodeType.SHUNT);
+                ((!node.isFictitious() && node.getType() != Node.NodeType.SHUNT) ||
+                        (node.isFictitious() && node.getComponentType() == ComponentType.THREE_WINDINGS_TRANSFORMER));
     }
 
-    private void incorporateComponents(Node node, Element g) {
+    private void incorporateComponents(Node node, Element g, SubstationDiagramStyleProvider styleProvider) {
         SVGOMDocument obj = componentLibrary.getSvgDocument(node.getComponentType());
         transformComponent(node, g);
         if (obj != null && canInsertComponentSVG(node)) {
-            insertComponentSVGIntoDocumentSVG(obj, g);
+            insertComponentSVGIntoDocumentSVG(obj, g, node, styleProvider);
         }
     }
 
-    private void insertComponentSVGIntoDocumentSVG(SVGOMDocument obj, Element g) {
+    private void insertComponentSVGIntoDocumentSVG(SVGOMDocument obj, Element g, Node node,
+                                                   SubstationDiagramStyleProvider styleProvider) {
         // The following code work correctly considering SVG part describing the component is the first child of "obj" the SVGDocument.
         // If SVG are written otherwise, it will not work correctly.
+        boolean firstCircle = true;
+        boolean secondCircle = true;
+        Optional<String> color;
 
         for (int i = 0; i < obj.getChildNodes().item(0).getChildNodes().getLength(); i++) {
             org.w3c.dom.Node n = obj.getChildNodes().item(0).getChildNodes().item(i).cloneNode(true);
+
+            if (n instanceof SVGCircleElement || n instanceof SVGEllipseElement) {
```

Comment:
It would be more generic if we agree in kind of SVG component spec that it has to be made of 3 parts modelled by 3 SVG groups that should be named something like WINDING1, WINDING2, WINDING3 instead of circles or elements.

Reply:
Modifications done